xapian-bindings/java/java.i

```diff
@@ -174,21 +174,63 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%ignore MSetIterator::next();
+%javamethodmodifiers MSetIterator::next_helper() "private"
+
 %extend MSetIterator {
-    Xapian::docid next () {
-	Xapian::docid tmp;
+    void next_helper () {
 	if (Xapian::iterator_valid(*self)) {
```

Comment:
I changed the helper function to just check the validity of the iterator and move on to the next item.